r2dbc-mysql/src/main/java/io/asyncer/r2dbc/mysql/QueryFlow.java

```diff
@@ -1311,12 +1319,12 @@ private static String buildStartTransaction(TransactionDefinition definition) {
             }
         }
 
-        if (readOnly != null) {
-            if (readOnly) {
-                builder.append(" READ ONLY");
-            } else {
-                builder.append(" READ WRITE");
+        if (Boolean.TRUE.equals(readOnly)) {
+            if (!first) {
+                builder.append(',');
             }
+
+            builder.append(" READ ONLY");
```

Comment:
I believe we may need to include the `READ WRITE` unless it is explicitly set to read-only.
without `READ WRITE`, it will fallback to the default mode, which differs from how it behaved previously.

Reply:
You are right, my mistake.

See [r2dbc-mariadb](https://github.com/mariadb-corporation/mariadb-connector-r2dbc/blob/master/src/main/java/org/mariadb/r2dbc/client/SimpleClient.java#L405), I'm writing something similar to it.

While it's not a big deal, considering that different distributions **may** have inconsistent implementations of `SET TRANSACTION READ ONLY/WRITE` , it might be better for us to continue using READ WRITE explicitly.